### PR TITLE
Add self count to FlameGraphDiff

### PR DESCRIPTION
**Pull request by LenkaNovak**
## Purpose 

This allows counting and displaying the allocation changes of individual functions (independent of their child nodes). The need for this was identified whilst attempting to quantify local performance in #205. In the future, this will allow us to isolate allocation changes within our software from changes in the upstream models.

This is tagged as part of #229 

## To-do

- [x] separate allocation calculation into child and current function contributions. 
- [x] implement in FlameGraphDiff plots
- [x] add a ci test 
- [x] add in docs

## QA
- [x] check buildkite plots are created
- [x] check ci tests pass

## Note
- this is still in a temporary format/location, until we test/determine how this would be useful for the rest of the CliMA team. 

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Review by valeriabarra — APPROVED** [review]

File: perf/flame_test.jl [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    test_func_name = "get_y.flame_test.jl.28"
+
+    # test flame diff
+    tracked_list["$test_func_name"] = 100
+    tracked_list["self_count_$test_func_name"] = 100
+    profile_data, new_tracked_list =
+        ProfileCanvasDiff.view(Profile.fetch(), tracked_list = tracked_list, self_count = false)
+    node = find_child(profile_data.data["all"], test_func_name, self_count = false)
+    @test node.count_change == node.count - 100
```
Comment (juliasloan25) [inline perf/flame_test.jl:99]: Why do you subtract 100 from `node.count` here? (And 50 from `node.count - child_sum` below?)

**Review by juliasloan25 — APPROVED** [review]
Looks good to me!

File: perf/ProfileCanvasDiff.jl [hunk 2]
```diff
@@ -206,17 +307,20 @@ function status(node::Profile.StackFrameTree, C::Bool)
     return st
 end
 
-function add_child(graph::ProfileFrame, node, C::Bool; tracked_list = Dict{String, Int}(;))
+function add_child(graph::ProfileFrame, node, C::Bool; tracked_list = Dict{String, Int}(;), self_count = false)
```
[the remark this answers is not in the document]
Comment (LenkaNovak) [reply perf/ProfileCanvasDiff.jl:310]: Well spotted! 👀  This was left from a previous implementation.

File: perf/flame_test.jl (hunk 1 above)
Comment (LenkaNovak) [reply perf/flame_test.jl:99]: In the first case the reference value (`tracked_list["$test_func_name"]`) is set to 100 counts and the `view()` function returns (amongst other things) the change between the reference file and the count. In the case where `self_count = true` the `view()` returns the difference between the reference independent count (here set to 50) and the current reference count (i.e., node.count - child_sum). I'll clarify this in a comment here.

**Pull request merged**